examples/xlio_ultra_api_ping_pong.c

```diff
@@ -423,74 +290,189 @@ static int create_polling_group(void)
-        return rc;
+    if (!g_state.api) {
+        return;
     }
 
-    printf("Server socket created\n");
-
-    /*
-     * Step 2: Bind socket to address
-     * ==============================
-     */
-    struct sockaddr_in addr = {0};
-    addr.sin_family = AF_INET;
-    addr.sin_port = htons(g_config.port);
-
-    if (g_config.ip) {
-        rc = inet_aton(g_config.ip, &addr.sin_addr);
-        if (rc == 0) {
-            fprintf(stderr, "Invalid IP address: %s\n", g_config.ip);
-            return -1;
+    /* Destroy client socket (server mode) */
+    if (g_state.client_sock) {
+        rc = g_state.api->xlio_socket_destroy(g_state.client_sock);
+        if (rc != 0) {
+            fprintf(stderr, "Failed to destroy client socket\n");
         }
-    } else {
-        addr.sin_addr.s_addr = INADDR_ANY;
-    }
 
-    rc = g_state.api->xlio_socket_bind(g_state.sock, (struct sockaddr *)&addr, sizeof(addr));
-    if (rc != 0) {
-        fprintf(stderr, "Failed to bind server socket: %s\n", strerror(errno));
-        return rc;
+        /* Wait for termination event */
+        while (!g_state.terminated) {
+            g_state.api->xlio_poll_group_poll(g_state.group);
+        }
     }
 
-    printf("Server socket bound to %s:%u\n", g_config.ip ? g_config.ip : "*", g_config.port);
-
-    /*
-     * Step 3: Start listening for connections
-     * =======================================
-     * Note: socket_accept_cb must be registered in the polling group for this to work.
-     */
+    /* Destroy main socket */
+    if (g_state.sock) {
+        g_state.terminated = false;
+        rc = g_state.api->xlio_socket_destroy(g_state.sock);
+        if (rc != 0) {
+            fprintf(stderr, "Failed to destroy socket\n");
+        }
+
+        /* Wait for termination event */
+        while (!g_state.terminated) {
+            g_state.api->xlio_poll_group_poll(g_state.group);
+        }
+    }
+
+    /* Destroy polling group */
+    if (g_state.group) {
+        rc = g_state.api->xlio_poll_group_destroy(g_state.group);
+        if (rc != 0) {
+            fprintf(stderr, "Failed to destroy polling group\n");
+        }
+    }
+
+    /* Cleanup RX accumulator */
+    if (g_state.rx_acc.data) {
+        free(g_state.rx_acc.data);
+        g_state.rx_acc.data = NULL;
+    }
+
+    /* Deregister and free memory for TX buffer pool */
+    cleanup_tx_buffer_pool();
+
+    /* Finalize XLIO */
+    rc = g_state.api->xlio_exit();
+    if (rc != 0) {
+        fprintf(stderr, "Failed to finalize XLIO\n");
+    }
+
+    printf("Resources cleaned up\n");
+}
+
+/*
+ * Send message using zero-copy operation with buffer pool
+ * =======================================================
+ * Performs the following steps:
+ *  1. Gets a buffer from the TX buffer pool
+ *  2. Copies the message
+ *  3. Adds a newline delimiter according to our protocol
+ *  4. The buffer pointer is stored in userdata_op for identification in completion callback
```

Comment:
all other steps are active, and this is passive. suggest to change:
"Stores the buffer pointer in userdata_op for completion callback identification"

Reply:
done